packages/js/src/components/contentAnalysis/Results.js

```diff
@@ -57,6 +65,46 @@ class Results extends Component {
 		}
 	}
 
+	/**
+	 * Factory method which creates a new instance of the default mark button.
+	 *
+	 * @param {string} ariaLabel 	The button aria-label.
+	 * @param {string} id 			The button id.
+	 * @param {string} className 	The button class name.
+	 * @param {string} status 		Status of the buttons. Supports: "enabled", "disabled", "hidden".
+	 * @param {function} onClick 	Onclick handler.
+	 * @param {boolean} isPressed 	Whether the button is in a pressed state.
+	 *
+	 * @returns {JSX.Element} A new mark button.
+	 */
+	createMarkButton( {
+		ariaLabel,
+		id,
+		className,
+		status,
+		onClick,
+		isPressed,
+	} ) {
+		return <Fragment>
+			<IconButtonToggle
+				marksButtonStatus={ status }
+				className={ className }
+				onClick={ onClick }
+				id={ id }
+				icon="eye"
+				pressed={ isPressed }
+				ariaLabel={ ariaLabel }
+			/>
+			{ this.props.shouldUpsellHighlighting && <div className="yst-root">
+				<Badge className="yst-absolute yst-px-[3px] yst-py-[3px] yst--right-[6.5px] yst--top-[6.5px]" size="small" variant="upsell">
+					<LockClosedIcon
+						className="yst-w-2.5 yst-h-2.5 yst-shrink-0" role="img" aria-hidden={ true } focusable={ false }
+					/>
```

Comment:
Normally, when we are using `LockClosedIcon` we are passing the aria attributes by using `useSvgAria()` hook from `@yoast/ui-library` package (for example [here](https://github.com/Yoast/wordpress-seo/blob/bbfd0653cbb0926171c0987acfad2976be0b258e/packages/js/src/components/modals/PremiumSEOAnalysisModal.js#L48)). However, we cannot use the hook in `createMarkButton` because `useSvgAria` is a hook which uses React `useMemo`. React hook cannot be called inside a condition (per [this documentation](https://legacy.reactjs.org/docs/hooks-rules.html)). In `AnalysisResult` component `createMarkButton` is rendered under a condition. Hence, if we were to use the `useSvgAria` inside `createMarkButton`, we would have an error.

Reply:
Ugh, that's unfortunate! 😕 Thanks for the context!